packages/flame/lib/src/widgets/animation_widget.dart

```diff
@@ -2,169 +2,62 @@ import 'dart:math';
+  }) : super(key: key);
 
-  SpriteAnimationWidget.asset({
-    required String path,
-    required SpriteAnimationData data,
-    Images? images,
-    this.playing = true,
-    this.anchor = Anchor.topLeft,
-    this.errorBuilder,
-    this.loadingBuilder,
-    Key? key,
-  })  : _animationFuture = (() => SpriteAnimation.load(
-              path,
-              data,
-              images: images,
-            )),
-        super(key: key);
+  /// The positioning [Anchor].
+  final Anchor anchor;
+
+  final SpriteAnimationController controller;
 
   @override
   Widget build(BuildContext context) {
-    return BaseFutureBuilder<SpriteAnimation>(
-      futureBuilder: _animationFuture,
-      builder: (_, spriteAnimation) {
-        return _SpriteAnimationWidget(
-          animation: spriteAnimation,
-          anchor: anchor,
-          playing: playing,
+    return AnimatedBuilder(
+      animation: controller,
+      builder: (_, __) {
+        return CustomPaint(
+          painter: SpritePainter(
+            controller.animation.getSprite(),
+            anchor,
+          ),
         );
       },
-      errorBuilder: errorBuilder,
-      loadingBuilder: loadingBuilder,
     );
   }
 }
 
-/// A [StatefulWidget] that render a [SpriteAnimation].
-class _SpriteAnimationWidget extends StatefulWidget {
-  /// The [SpriteAnimation] to be rendered
-  final SpriteAnimation animation;
-
-  /// The positioning [Anchor]
-  final Anchor anchor;
-
-  /// Should the [animation] be playing or not
-  final bool playing;
-
-  const _SpriteAnimationWidget({
+class SpriteAnimationController extends AnimationController {
+  SpriteAnimationController({
+    required TickerProvider vsync,
     required this.animation,
-    this.playing = true,
-    this.anchor = Anchor.topLeft,
-  });
-
-  @override
-  State createState() => _SpriteAnimationWidgetState();
-}
-
-class _SpriteAnimationWidgetState extends State<_SpriteAnimationWidget>
-    with SingleTickerProviderStateMixin {
-  AnimationController? _controller;
-  double? _lastUpdated;
-
-  @override
-  void initState() {
-    super.initState();
-    widget.animation.onComplete = _pauseAnimation;
-    _setupController();
-    if (widget.playing) {
-      _initAnimation();
-    }
-  }
-
-  @override
-  void didUpdateWidget(_SpriteAnimationWidget oldWidget) {
-    super.didUpdateWidget(oldWidget);
-
-    if (oldWidget.animation != widget.animation) {
-      oldWidget.animation.onComplete = null;
-      _setupController();
-    }
-
-    if (widget.playing) {
-      _initAnimation();
-    } else {
-      _pauseAnimation();
-    }
-  }
-
-  void _initAnimation() {
-    setState(() {
-      widget.animation.reset();
-      _lastUpdated = DateTime.now().millisecond.toDouble();
-      _controller?.repeat(
-        // Approximately 60 fps
-        period: const Duration(milliseconds: 16),
-      );
-    });
+  }) : super(vsync: vsync) {
+    duration = Duration(seconds: animation.totalDuration().ceil());
```

Comment:
The `ceil` makes the total duration to be approximated. I still need to define an algorithm to calculate the `totalDuration` as `Duration`, without losing precision.